### ISSUE TITLE
Complier can't introduce a list of completions for `Effect<Action>` when using `BindingReducer()` in body

### ISSUE THREAD

**Issue by ValseLee** [pointfreeco/swift-composable-architecture#2386]
### Description

- I tried to find any other existing issues or discussion with keyword 'completion' but there is no single one. So I make this issue with a `bug` Label.
- If I use `BindingReducer()` in `body`, compiler stops introducing a list of method completions in `Effect<Action>`.
- I attached some code blocks and screen captures below.

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

- Compiler can introduce list of method completions of `Effect<Action>`.
- Compiler can introduce list of method completions of `Reduce<State, Action>`.

### Actual behavior

- Compiler **can't introduce** list of method completions of `Effect<Action>`.
- Compiler **can introduce** list of method completions of `Reduce<State, Action>`.

### Steps to reproduce

1. When using `BindingReducer()`
```swift
struct ForIssue: Reducer {
  struct State: Equatable { var id: UUID }
  enum Action: Equatable, BindableAction {
    case binding(BindingAction<State>)
    case onButtonTapped
  }
  
  var body: some ReducerOf<Self> {
    /// if BindingReducer() exists in `body`,
    /// compiler can't introduce list of completions
    BindingReducer()

    Reduce { state, action in
      switch action {
      case .binding:
        return .none
        
      case .onButtonTapped:
        return .none
      }
    }
  }
}
```
![image](https://github.com/pointfreeco/swift-composable-architecture/assets/82270058/d2fa52ac-371f-4b0d-9479-280b4d73fcdc)

2. When stop using `BindingReducer()`
```swift
struct ForIssue: Reducer {
  struct State: Equatable { var id: UUID }
  enum Action: Equatable, BindableAction {
    case binding(BindingAction<State>)
    case onButtonTapped
  }
  
  var body: some ReducerOf<Self> {
    /// if BindingReducer() exists in `body`,
    /// compiler can't introduce list of completions
    /// BindingReducer()

    Reduce { state, action in
      switch action {
      case .binding:
        return .none
        
      case .onButtonTapped:
        return .none
      }
    }
  }
}
```
![image](https://github.com/pointfreeco/swift-composable-architecture/assets/82270058/b09a0186-ba51-4c9a-8c0b-910f47cfada2)

3. `Reduce<State, Action>` method completions are working as normal, if `body` has `BidningReducer()` or not.
![image](https://github.com/pointfreeco/swift-composable-architecture/assets/82270058/c4d8904e-74ce-4178-8139-508d95802c9b)

### The Composable Architecture version information

beta 1.0.0

### Destination operating system

iOS 16.2

### Xcode version information

14.2 (14C18)

### Swift Compiler version information

```shell
swift-driver version: 1.62.15 Apple Swift version 5.7.2 (swiftlang-5.7.2.135.5 clang-1400.0.29.51)
Target: arm64-apple-macosx13.0
```

**Comment by nivanchikov** [comment]
@ValseLee can you try this tip by specifying the generics of your Reduce reducer: `Reduce<State, Action> { state, action in … }` as per this comment by @tgrapperon:
https://github.com/pointfreeco/swift-composable-architecture/discussions/1666#discussioncomment-4212335

It generally helped me in similar situations.

**Comment by ValseLee** [comment]
@nivanchikov So.... I didn't try your solution before, but it is perfectly working!
Thank you for your kind explanation with #1666 !

**Comment by stephencelis** [comment]
@ValseLee This was fixed in a later Xcode, as well. If you can upgrade from 14.2, which was released last year, to 14.3.1, this should be fixed without the need of those generics!

**Comment by JeongAYoo** [comment]
@stephencelis  I'm working on Xcode 15.0.1 with TCA 1.9.2 but I have the exact same problem described above...
Is there something to check in project's build settings?
